workers/datasets_based/src/datasets_based/workers/first_rows.py

```diff
@@ -480,6 +506,24 @@ def compute_first_rows_response(
             raise FeaturesError("The split features (columns) cannot be extracted.", cause=err) from err
     else:
         features = info.features
+
+    if len(features) > columns_max_number:
+        raise TooManyColumnsError(
+            f"Too many columns. The maximum supported number of columns is {columns_max_number}."
+        )
+
+    # validate size of response without the rows
+    features_list = to_features_list(features=features)
+    response: FirstRowsResponse = {
```

Comment:
Good idea.

I would prefer to give it another name though, just in case, to avoid returning this. eg

```suggestion
    response_features_only: FirstRowsResponse = {
```